extension/js/common/helpers.ts

```diff
@@ -164,3 +164,11 @@ export const processAndStoreKeysFromEkmLocally = async (
   }
   return { updateCount: encryptedKeys?.keys.length ?? 0 + (existingKeys.length - keysToRetain.length), noKeysSetup: !(encryptedKeys?.keys.length || keysToRetain.length) };
 };
+
+export const getLocalKeyExpiration = async ({ acctEmail }: Bm.GetLocalKeyExpiration): Promise<Bm.Res.GetLocalKeyExpiration> => {
+  const kis = await KeyStore.get(acctEmail);
+  const parsedKeys = await Promise.all(kis.map(async (ki) => await KeyUtil.parse(ki.public)));
+  const keyExpiration = parsedKeys
+    .sort((key1, key2) => (key1.expiration ?? Number.MAX_SAFE_INTEGER) < (key2.expiration ?? Number.MAX_SAFE_INTEGER) ? 1 : -1)[0]?.expiration;
```

Comment:
I suggest to simplify this: we don't need the sorted array, only the `min` value?
We can simply `.map(key => key.expiration ?? Number.MAX_SAFE_INTEGER)` (or during parsing `(await KeyUtil.parse(ki.public)).expiration ?? Number.MAX_SAFE_INTEGER`
and then find the min value using Math.min or Array.prototype.reduce
as shown in these examples https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max

Reply:
expirations.reduce((a, b) => Math.min(a, b), 0);

Reply:
or even without mapping (though in our case mapping would simplify the code) as the reduced valued type need not match the array element types:
`const keyExpiration = parsedKeys.reduce((expiration, key2) => Math.min(expiration, key2.expiration ?? Number.MAX_SAFE_INTEGER), parsedKeys[0].expiration ?? Number.MAX_SAFE_INTEGER);`